consumer.go

```diff
@@ -192,19 +192,24 @@ func (c *Consumer) source(
 		return int(m.genLine) >= genLine
 	})
 
-	// Mapping not found.
+	var match *mapping
+	// Mapping not found
 	if i == len(m.mappings) {
-		return
-	}
-
-	match := &m.mappings[i]
-
-	// Fuzzy match.
-	if int(match.genLine) > genLine || int(match.genColumn) > genColumn {
-		if i == 0 {
+		// lets see if the line is correct but the column is bigger
+		match = &m.mappings[i-1]
```

Comment:
This does not look right. It looks like you want to change https://github.com/go-sourcemap/sourcemap/blob/73a0ee2ad6086891e4fcf116763e2a239693cfea/consumer.go#L187-L193 instead.

Reply:
well, I guess we can move it there as well.

The thing is that `sort.Search` actually *correctly* search and finds that the place where this should be is *at the end* of the slice. In general `sort.Search` just finds where a thing is supposed to be not exactly if it's *there*, it's more answering the question where an element should be added in that regard. 

This is a kin to the fuzzy search below where if what it was asked for isn't exactly there it will get the previous one if possible

Reply:
You are right. Probably I should not have used `sort.Search` here, because there are too many corner cases.